core/state/statedb.go

```diff
@@ -211,7 +212,12 @@ func (self *StateDB) UpdateStateObject(stateObject *StateObject) {
 		self.db.Put(stateObject.CodeHash(), stateObject.code)
 	}
```

Comment:
This `if` is interesting, because in the original code already the length check was for the hash and not the code itself. So we've essentially always pushed in the empty code into the database as the length *was* actually that of the hash, hence always `>0`. Shouldn't we check against the actual `len(stateObject.code)` instead to figure out whether we `Put` or not?